aws/terraform/modules/metaflow/modules/metadata-service/ec2.tf

```diff
@@ -0,0 +1,111 @@
+resource "aws_security_group" "metadata_service_security_group" {
+  name        = local.metadata_service_security_group_name
+  description = "Security Group for Fargate which runs the Metadata Service."
+  vpc_id      = var.metaflow_vpc_id
+
+  ingress {
+    from_port   = 8080
+    to_port     = 8080
+    protocol    = "tcp"
+    cidr_blocks = [var.vpc_cidr_block]
+    description = "Allow API calls internally"
+  }
+
+  ingress {
+    from_port   = 8082
+    to_port     = 8082
+    protocol    = "tcp"
+    cidr_blocks = [var.vpc_cidr_block]
+    description = "Allow API calls internally"
+  }
+
+  ingress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = -1
+    self        = true
+    description = "Internal communication"
+  }
+
+  # egress to anywhere
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1" # all
+    cidr_blocks = ["0.0.0.0/0"]
+    description = "Allow all external communication"
+  }
+
+  tags = merge(
+    var.standard_tags,
+    {
+      Metaflow = "true"
+    }
+  )
+}
+
+resource "aws_lb" "this" {
+  name               = "${var.resource_prefix}nlb${var.resource_suffix}"
+  internal           = true
+  load_balancer_type = "network"
+  subnets            = [var.subnet1_id, var.subnet2_id]
+
+  tags = var.standard_tags
+}
+
+resource "aws_lb_target_group" "this" {
+  name        = "${var.resource_prefix}trgtgrp${var.resource_suffix}"
+  port        = 8080
+  protocol    = "TCP"
+  target_type = "ip"
+  vpc_id      = var.metaflow_vpc_id
+
+  health_check {
+    protocol            = "TCP"
+    interval            = 10
+    healthy_threshold   = 2
+    unhealthy_threshold = 2
+  }
+
+  tags = var.standard_tags
+}
+
+resource "aws_lb_target_group" "db_migrate" {
+  name        = "${var.resource_prefix}dbtrgtgrp${var.resource_suffix}"
```

Comment:
This might be a cascading issue that affects other stuff, but I'm going to draw attention to it here since I've hit it twice.  Certain resources, target groups included, have a 32 character limit.  The baseline example.tfvars and this naming convention result in a name that ends up being 33 characters.  I fix it by changing this line to `"${var.resource_prefix}dbtg${var.resource_suffix}"`.  We should definitely ensure that the template works out-of-the-box with the provided variables either by shortening this line or providing shorter variable defaults, but it's still worth calling out that these naming conventions teeter dangerously close to character limits, and even a wee bit of user-fiddling in the labeling could result in overflow.  It's not a deal-breaker, but I'd anticipate running into it in the wild occasionally.